### ISSUE TITLE
Copying to same folder should automatically rename

### ISSUE THREAD

**Issue by tgrelka** [files-community/Files#3900]
**Is your feature request related to a problem? Please describe.**
Sometimes, I need to make a copy of a file to change the extenison (e.g. fom .txt to .csv to view a table) or make a backup.
Most of the time this happens in the same folder. It is quite annyoing that each time the dialog pops up asking me if I want to replace the file or create a new name.

**Describe the solution you'd like**
Since the names are checked anyway, I’d like to see a full path check implemented, which instead of asking to replace the file recognizes that source and destination folder are identical and instead of asking to replace the file (which makes no sense in that case - why would I want to replace the file I just copied?) just automatically generates a new name.

Windows explorer does it this way, automatically generating a new file „... - Copy“ when source and dest are the same, otherwise asking if I want to replace the file or not.

**Comment by gave92** [comment]
The biggest issue is probably that a dialog is shown for every item and there is no "do the same for others" option

**Comment by tgrelka** [comment]
No, often I just copy a single item. The issue is the dialogue appearing at all when the intent is clear and obvious.
Copying multiple files from one location to another should have a „Do the same for all items“ option, like Windows Explorer has, but that’s a different issue.
